### ISSUE TITLE
resolve templates in env section

### ISSUE THREAD

**Issue by troian** [goreleaser/goreleaser#1820]
In some cases like [cross-compiling](https://gophers.slack.com/archives/C3T1JU24B/p1600905690004300) it is quite handy to substitute long strings especially when there are repetitive.
In [example](https://github.com/troian/golang-cross/blob/master/.goreleaser.yaml) instead of writing
```yaml
builds:
  - id: darwin-amd64
    env:
      - PKG_CONFIG_SYSROOT_DIR=/go/src/github.com/troian/golang-cross/sysroot/macos/amd64
      - PKG_CONFIG_PATH=/go/src/github.com/troian/golang-cross/sysroot/macos/amd64/usr/local/lib/pkgconfig
  - id: linux-amd64
    env:
      - CGO_ENABLED=1
      - PKG_CONFIG_SYSROOT_DIR=/go/src/github.com/troian/golang-cross/sysroot/linux
      - PKG_CONFIG_PATH=/go/src/github.com/troian/golang-cross/sysroot/linux/usr/lib/x86_64-linux-gnu/pkgconfig
```
it would be good to use path prefixes from env variable
```yaml
builds:
  - id: darwin-amd64
    env:
      - PKG_CONFIG_SYSROOT_DIR={{ .Env.SYSROOT_DIR }}/macos/amd64
      - PKG_CONFIG_PATH={{ .Env.SYSROOT_DIR }}/macos/amd64/usr/local/lib/pkgconfig
  - id: linux-amd64
    env:
      - CGO_ENABLED=1
      - PKG_CONFIG_SYSROOT_DIR={{ .Env.SYSROOT_DIR }}/linux
      - PKG_CONFIG_PATH={{ .Env.SYSROOT_DIR }}/linux/usr/lib/x86_64-linux-gnu/pkgconfig
```

**Comment by m4ver1k** [comment]
@caarlos0 can I give this a try ?

**Comment by caarlos0** [comment]
hey @m4ver1k , yeah go ahead! Thanks 🙏

**Comment by m4ver1k** [comment]
Hi @caarlos0,
have made the changes, but the snapcraft test keeps failing, I'm on mac and installed snapcraft using brew.

```--- FAIL: TestRunPipeWithName (0.58s)
    snapcraft_test.go:141:
        	Error Trace:	snapcraft_test.go:141
        	Error:      	Received unexpected error:
        	            	failed to generate snap package: Snapping...
        	            	Sorry, an error occurred in Snapcraft:
        	            	Failed to create snap, snap command failed:
        	            	stdout:

        	            	stderr:
        	            	2020/10/07 00:30:13.487491 container.go:213: in snap "testsnapname": path "mybin" does not exist
        	            	error: cannot pack "/var/folders/0z/y2vh6zjj0m92p065bzb5w3x00000gp/T/archivetest076617848/dist/foo_amd64/prime": snap is unusable due to missing files

        	            	Traceback (most recent call last):
        	            	  File "/usr/local/bin/snapcraft", line 8, in <module>
        	            	    sys.exit(run())
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/click/core.py", line 829, in __call__
        	            	    return self.main(*args, **kwargs)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/click/core.py", line 782, in main
        	            	    rv = self.invoke(ctx)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
        	            	    return _process_result(sub_ctx.command.invoke(sub_ctx))
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/snapcraft/cli/_command.py", line 88, in invoke
        	            	    return super().invoke(ctx)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
        	            	    return ctx.invoke(self.callback, **ctx.params)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/click/core.py", line 610, in invoke
        	            	    return callback(*args, **kwargs)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/snapcraft/cli/lifecycle.py", line 372, in pack
        	            	    _pack(directory, output=output)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/snapcraft/cli/lifecycle.py", line 220, in _pack
        	            	    snap_filename = _run_pack(command)
        	            	  File "/usr/local/Cellar/snapcraft/4.3/libexec/lib/python3.8/site-packages/snapcraft/cli/lifecycle.py", line 189, in _run_pack
        	            	    raise RuntimeError(
        	            	RuntimeError: Failed to create snap, snap command failed:
        	            	stdout:

        	            	stderr:
        	            	2020/10/07 00:30:13.487491 container.go:213: in snap "testsnapname": path "mybin" does not exist
        	            	error: cannot pack "/var/folders/0z/y2vh6zjj0m92p065bzb5w3x00000gp/T/archivetest076617848/dist/foo_amd64/prime": snap is unusable due to missing files

        	            	We would appreciate it if you created a bug report at
        	            	https://launchpad.net/snapcraft/+filebug with the above text included.
        	            	You can find the traceback in file '/var/folders/0z/y2vh6zjj0m92p065bzb5w3x00000gp/T/tmpb163ww_m/trace.txt'.
        	Test:       	TestRunPipeWithName
FAIL
FAIL	github.com/goreleaser/goreleaser/internal/pipe/snapcraft	2.748s
FAIL
```

Could you please point me in right direction.

**Comment by caarlos0** [comment]
yeah, happens here too, probably related to the snapcraft version... probably need to fix our tests.

**Comment by caarlos0** [comment]
created https://github.com/goreleaser/goreleaser/issues/1843

**Comment by m4ver1k** [comment]
@caarlos0 please review the draft PR and let me know if you have any feedbacks, will do one more round of testing and raise final PR.

https://github.com/goreleaser/goreleaser/pull/1845

**Comment by caarlos0** [comment]
fixed in #2246
